boto3/session.py

```diff
@@ -174,14 +175,21 @@ def client(self, service_name, region_name=None, api_version=None,
         :param aws_session_token: The session token to use when creating
             the client.  Same semantics as aws_access_key_id above.
 
+        :type config: botocore.client.Config
+        :param config: Advanced client configuration options. If region_name
+            is specified in the client config, its value will take precedence
+            over environment variables and configuration values, but not over
+            a region_name value passed explicitly to the method.
+
         :return: Service client instance
+
         """
         return self._session.create_client(
             service_name, region_name=region_name, api_version=api_version,
             use_ssl=use_ssl, verify=verify, endpoint_url=endpoint_url,
             aws_access_key_id=aws_access_key_id,
             aws_secret_access_key=aws_secret_access_key,
-            aws_session_token=aws_session_token)
+            aws_session_token=aws_session_token, config=config)
 
     def resource(self, service_name, region_name=None, api_version=None,
```

Comment:
Yeah it seems we either add the client config arg to `resource()`, or we allow a user to specify a client object when creating a resource.  The latter seems more general and would allow you to use a single client across resources which could be nice.

If we do allow for a client config to be passed into `resource()`, what should be the behavior if the user already provided a non-None `user_agent_extra`?